components/WeatherDisplay.js

```diff
@@ -12,7 +12,7 @@ const API = process.env.NEXT_PUBLIC_WEATHER_API_KEY;
 
 export default function WeatherDisplay({ area, lat, lng }) {
   const { data: weather } = useSWR(
-    `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lng}&units=metric&appid=${API}`
+    `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lng}&units=metric&appid=${API}`,
```

Comment:
Is this already fixed by the PR of Dominik? if so, please merge the main branch into this feature branch to be up to date. if not, please change this code such that the API Key is not exposed to the frontend.

Reply:
Ah okay, great suggestion. We got it on the plate. One weather fetching situation slippe through our fingers, but Dominik will create a new UserStory for that so everything is in order. For this specific PR i would still leave it in as the weather fetching changes are still in QA and will be merged before this PR.